js/tc39/tc39_test.go

```diff
@@ -647,6 +658,59 @@ func (ctx *tc39TestCtx) runTC39Script(name, src string, includes []string, vm *g
 	return early, origErr, err
 }
 
+func (ctx *tc39TestCtx) runTC39Module(name, src string, includes []string, vm *goja.Runtime) (early bool, origErr, err error) {
+	currentFS := os.DirFS(".")
+	if err != nil {
+		panic(err)
+	}
+	moduleRuntime := modulestest.NewRuntime(ctx.t)
+	moduleRuntime.VU.RuntimeField = vm
+	early = true
+	err = ctx.runFile(ctx.base, path.Join("harness", "assert.js"), vm)
+	if err != nil {
+		return
+	}
+
+	err = ctx.runFile(ctx.base, path.Join("harness", "sta.js"), vm)
+	if err != nil {
+		return
+	}
+
+	for _, include := range includes {
+		err = ctx.runFile(ctx.base, path.Join("harness", include), vm)
+		if err != nil {
+			return
+		}
+	}
+
+	comp := ctx.compilerPool.Get()
+	defer ctx.compilerPool.Put(comp)
+	comp.Options = compiler.Options{Strict: false, CompatibilityMode: lib.CompatibilityModeExtended}
+
+	mr := modules.NewModuleResolver(nil,
+		func(specifier *url.URL, name string) ([]byte, error) {
+			return fs.ReadFile(currentFS, specifier.Path[1:])
+		},
+		comp)
+	u, err := url.Parse(path.Join(ctx.base, name))
+	if err != nil {
+		panic(err) // TODO fix
```

Comment:
Non-blocking: Out of curiosity, can we give more details for future fixing?

Reply:
Like fb139a484